### ISSUE TITLE
Action payload optional prevents object destructuring

### ISSUE THREAD

**Issue by mrkswrnr** [ClickerMonkey/vuex-typescript-interface#7]
The definition of [ActionHandler](https://github.com/ClickerMonkey/vuex-typescript-interface/blob/61cf3dc48b1867775fab155c4a1c1d5442860ca5/index.d.ts#L163) marks payload as optional. This makes payload destructuring harder.

> A mutation/action could require a payload, but at the moment it will appear optional.

Why did you choose this approach?

Example:
```typescript
interface IStore
{
  fetch (payload: {id: number}): Promise<void>
}

const store = new Store<IStore>({
  actions: {
    // Property 'id' does not exist on type '{ id: number; } | undefined'. # ts(2339)
    async fetch ({ commit }, {id}) {
      // action
    }
  }
});
```

**Comment by ClickerMonkey** [comment]
Thanks for bringing this to my attention, working on it.

**Comment by ClickerMonkey** [comment]
Also, I chose this approach because when you're invoking an action/commit it might not have a payload, so to require one would be silly. I'll see if I can get proper payload existence detection working, and if possible I can get this working properly.

**Comment by mrkswrnr** [comment]
Thank you for working on it!

> Also, I chose this approach because when you're invoking an action/commit it might not have a payload, so to require one would be silly. 

I understand. I tried to come up with a conditional type that casts the payload parameter to void/undefined but while Typescript accepts omitting void/undefined parameters when declaring them directly, this does not seems to be working when using conditional types. 

If there is no way to omit the parameter I would prefer to be forced to use an empty object as parameter instead of not being able to use object destructuring. But of course this is your decision as a module author.

**Comment by ClickerMonkey** [comment]
Resolved with v1.0.2
